src/compute-client/src/controller/instance.rs

```diff
@@ -715,6 +721,9 @@ where
             self.update_read_capabilities(&mut compute_read_capability_changes)
                 .await?;
         }
+        if !dropped_collection_ids.is_empty() {
+            self.update_dropped_collections(dropped_collection_ids)?;
+        }
```

Comment:
I *think* you'll have to move this before the `update_read_capabilities` call. The reason is that `update_read_capabilities` also can invoke `update_dropped_collections`, which might remove some collections that are present in `dropped_collection_ids`. So when you call `update_dropped_collections` here again, it might error because it can't find the specified collection.

Alternatively you could make `update_dropped_collections` simply ignore collections that don't exist. That's probably the more robust solution.

Reply:
Good catch, I am going ahead with the latter suggestion to make the code more robust.